### ISSUE TITLE
issues with dem_proc color-relief

### ISSUE THREAD

**Issue by mdsumner** [USDAForestService/gdalraster#428]
##  1

I can't get this to work: 

```R
sq <- seq(2400, 3100, by = 100)
col <- col2rgb(hcl.colors(length(sq)))
writeLines(paste(sq, col[1, ], col[2, ], col[3, ]), pal <- tempfile(fileext = ".txt"))
library(gdalraster)
lcp_file <- system.file("extdata/storm_lake.lcp", package="gdalraster")
dem_proc("color_relief", lcp_file, tf <- tempfile(fileext = ".tif"), color_file = pal)
# ERROR 5: Invalid processing
# Error: DEM processing failed
```

At the cli I get what I expect: 

```R
system(sprintf("gdaldem color-relief %s %s %s -b 1", lcp_file, pal, tf))
plot_raster(ds <- new(GDALRaster, tf), nbands = 3)
ds$close()
```

![image](https://github.com/USDAForestService/gdalraster/assets/4107631/c08e2dbd-b6ef-46ab-9543-534e07763cad)

## 2 

I see that it's "color_relief" recognized by dem_proc and not "color-relief". If that's intended it's not documented as such (apart from being used as a name at the R level). 

Thanks!

**Comment by mdsumner** [comment]
oh, I see that's actually the bug. Using "color_relief" to look up the properties, and passing that in rather than "color-relief". 

Suggested fix at 

https://github.com/USDAForestService/gdalraster/blob/c5ef54c37be515ed3921ef1b35687c7b5904aee8/R/gdalraster_proc.R#L52

quote the name: 

```
 "color-relief" = character(),
```

and for backwards-compatibility check for "color_relief" on input (but, did it ever work for anyone? - maybe just change it). Thanks!!

**Comment by ctoney** [comment]
Thanks! Fixed in #430.
